config/config.go

```diff
@@ -503,6 +513,30 @@ func New(v *viper.Viper) (*Configuration, error) {
 		c.GDPR.NonStandardPublisherMap[c.GDPR.EEACountries[i]] = s
 	}
 
+	// To look for a purpose's vendor exceptions in O(1) time, for each purpose we fill this hash table located in the
+	// VendorExceptions field of the GDPR.TCF2.PurposeX struct defined in this file
+	purposeConfigs := []*Purpose{
```

Comment:
Why do the Purpose elements of this slice need to be a reference type?

Reply:
I made this a reference type because I didn't want there to be two versions of each purpose config, one being exactly what's specified by the host and the second being a copy of what's specified by the host plus the generated `VendorExceptionMap` hash set.
The permissions object constructor takes an instance of `config.GDPR` as a parameter. To reduce confusion I thought continuing the pass the lone instance of `config.GDPR` configured by the host plus the generated hash set was the simplest solution. Alternatively, we could copy all of `config.GDPR` and add the hash set to it but I thought that was dirty and complicated things.
I was also trying to avoid changing the constructor signature at this time with additional refactors on the horizon.